src/IR/IR.hs

```diff
@@ -315,49 +320,72 @@ wellFormed (Prim   p    es   _) = wfPrim p es && all wellFormed es
   wfPrim (PrimOp PrimLe    ) [_, _]    = True
   wfPrim _                   _         = False
 
+{- | Pretty Print the IR
+
+src/IR.hs:          fmtPretty   
+
+                        |
+                        |
+                        v
+
+src/IR/IR.hs:          fmt       // FMT type class
+
+                        |
+                        |
+                        v
+
+src/IR/IR.hs:         pretty     // Pretty type class
+                                 
+                        |        
+                        |        
+                        v        
+
+                       AST
+
+-}
+
+-- | Pretty Typeclass: translates IR into a one to one Doc representation
 instance Pretty t => Pretty (Program t) where
   pretty Program { programDefs = ds } = vsep $ map pretty ds
-  -- TODO: type defs
   -- TODO: how to represent entry point?
 
+-- | pretty print a binder, which is an alias for Maybe Varid
+prettyBinder :: Binder -> Doc ann
+prettyBinder (Just varid) = pretty varid
+prettyBinder _            = pretty '_'
+
 instance Pretty t => Pretty (Expr t) where
```

Comment:
Could you change this instance to `instance Pretty (Expr ())`? I want the fact that type annotations are being omitted to be visible in the type signature